DEVELOPER.md

```diff
@@ -1,9 +1,6 @@
 # Developer Notes
 =================
 
-## Getting started
```

Comment:
I feel like this section is covered by the [CONTRIBUTING.md](https://github.com/m3db/m3db/blob/master/CONTRIBUTING.md) file.

Reply:
Sounds good